pkg/flow/internal/controller/config_export.go

```diff
@@ -0,0 +1,82 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/grafana/agent/pkg/river/ast"
+	"github.com/grafana/agent/pkg/river/diag"
+	"github.com/grafana/agent/pkg/river/vm"
+)
+
+type ExportConfigNode struct {
+	configNode      sharedBlockNode
+	onExportsChange func(exports map[string]any) // Invoked when the managed component updated its exports
+}
+
+var _ BlockNode = (*ExportConfigNode)(nil)
+
+// NewExportConfigNode creates a new ExportConfigNode from an initial ast.BlockStmt.
+// The underlying config isn't applied until Evaluate is called.
+func NewExportConfigNode(block *ast.BlockStmt, globals ComponentGlobals, isInModule bool) (*ExportConfigNode, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	if !isInModule {
+		diags.Add(diag.Diagnostic{
+			Severity: diag.SeverityLevelError,
+			Message:  "export blocks only allowed inside a module",
+			StartPos: ast.StartPos(block).Position(),
+			EndPos:   ast.EndPos(block).Position(),
+		})
```

Comment:
Should this be returning so the node doesn't get created anyway?

Reply:
good catch on these, will do